### DISCUSSION TITLE
An exception occurred in driver: SQLSTATE[HY000] [2002] Permission denied

### DISCUSSION THREAD

**Discussion by tapli** [pimcore/pimcore#14697] [Q&A / Support] [answered]
I have installed Pimcore on Ubuntu using the Pimcore installation guide. I have verified that I can connect to MySQL PimCore database using the pimcore specific user id and password. Yet I am getting the "Permission denied" error.  I also verified that MySQL is not running in a safe mode. The database.yaml file under /config/local directory has the correct MySQL settings as well. I have also restarted MySQL and Apache2. 
What am I missing here?

**Comment by martineiber** [comment, marked as answer]
Hey @tapli,

Do you get this error during the installation or after the installation?
Could you share your database.yaml file?
Does your `server_version` in your database.yaml match your current MySQL version?

**Reply by tapli** [reply]
Hi @martineiber 

I did not get an error during the installation except for the one where Pimcore could not drop and create a function. I do not believe that is causing this issue. I checked the server_version parameter and it is matching with the MySQL version

mysql  Ver 8.0.32-0ubuntu0.22.04.2 for Linux on x86_64 ((Ubuntu))

[database.zip](https://github.com/pimcore/pimcore/files/11018216/database.zip)

I changed the port number from 3306 to 8899. And that did not make any difference.

**Reply by tapli** [reply]
@martineiber 

Here is the log message stack trace.

Uncaught PHP Exception Doctrine\DBAL\Exception\ConnectionException: "An exception occurred in driver: SQLSTATE[HY000] [2002] Permission denied" at /project/my-project/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 112
request Show context Hide trace
{▼
  [/project/my-project/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:112](http://localhost/_profiler/open?file=vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php&line=112#line112) {▼
    [Doctrine\DBAL\Driver\AbstractMySQLDriver->convertException($message, DriverException $exception)](http://localhost/_profiler/open?file=vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php&line=47#line47) …
    › case '2005':
    ›    [ return new ConnectionException($message, $exception);](http://localhost/_profiler/open?file=vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php&line=112#line112)
    › 
  }
  [/project/my-project/vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php:182](http://localhost/_profiler/open?file=vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php&line=182#line182) {▶}
  [/project/my-project/vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php:169](http://localhost/_profiler/open?file=vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php&line=169#line169) {▶}
  [/project/my-project/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOMySql/Driver.php:31](http://localhost/_profiler/open?file=vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOMySql/Driver.php&line=31#line31) {▶}
  [/project/my-project/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:412](http://localhost/_profiler/open?file=vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php&line=412#line412) {▶}
  [/project/my-project/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:1952](http://localhost/_profiler/open?file=vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php&line=1952#line1952) {▶}
  [/project/my-project/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:1290](http://localhost/_profiler/open?file=vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php&line=1290#line1290) {▶}
  [/project/my-project/vendor/pimcore/pimcore/lib/Db/PimcoreExtensionsTrait.php:92](http://localhost/_profiler/open?file=vendor/pimcore/pimcore/lib/Db/PimcoreExtensionsTrait.php&line=92#line92) {▶}
  [/project/my-project/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:684](http://localhost/_profiler/open?file=vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php&line=684#line684) {▶}
  [/project/my-project/vendor/pimcore/pimcore/models/Tool/SettingsStore/Dao.php:79](http://localhost/_profiler/open?file=vendor/pimcore/pimcore/models/Tool/SettingsStore/Dao.php&line=79#line79) {▶}
  [/project/my-project/vendor/pimcore/pimcore/models/Tool/SettingsStore.php:131](http://localhost/_profiler/open?file=vendor/pimcore/pimcore/models/Tool/SettingsStore.php&line=131#line131) {▶}
  [/project/my-project/vendor/pimcore/pimcore/lib/Extension/Bundle/Installer/SettingsStoreAwareInstaller.php:153](http://localhost/_profiler/open?file=vendor/pimcore/pimcore/lib/Extension/Bundle/Installer/SettingsStoreAwareInstaller.php&line=153#line153) {▶}
  [/project/my-project/vendor/pimcore/pimcore/lib/Extension/Bundle/PimcoreBundleManager.php:583](http://localhost/_profiler/open?file=vendor/pimcore/pimcore/lib/Extension/Bundle/PimcoreBundleManager.php&line=583#line583) {▶}
  [/project/my-project/vendor/pimcore/pimcore/lib/Extension/Bundle/PimcoreBundleManager.php:114](http://localhost/_profiler/open?file=vendor/pimcore/pimcore/lib/Extension/Bundle/PimcoreBundleManager.php&line=114#line114) {▶}
  [/project/my-project/vendor/pimcore/pimcore/lib/Extension/Bundle/PimcoreBundleManager.php:673](http://localhost/_profiler/open?file=vendor/pimcore/pimcore/lib/Extension/Bundle/PimcoreBundleManager.php&line=673#line673) {▶}
  [/project/my-project/vendor/pimcore/pimcore/lib/Extension/Bundle/PimcoreBundleManager.php:622](http://localhost/_profiler/open?file=vendor/pimcore/pimcore/lib/Extension/Bundle/PimcoreBundleManager.php&line=622#line622) {▶}
  [/project/my-project/vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/LoginController.php:302](http://localhost/_profiler/open?file=vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/LoginController.php&line=302#line302) {▶}
  [/project/my-project/vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/LoginController.php:107](http://localhost/_profiler/open?file=vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/LoginController.php&line=107#line107) {▶}
  [/project/my-project/vendor/symfony/http-kernel/HttpKernel.php:163](http://localhost/_profiler/open?file=vendor/symfony/http-kernel/HttpKernel.php&line=163#line163) {▶}
  [/project/my-project/vendor/symfony/http-kernel/HttpKernel.php:75](http://localhost/_profiler/open?file=vendor/symfony/http-kernel/HttpKernel.php&line=75#line75) {▶}
  [/project/my-project/vendor/symfony/http-kernel/Kernel.php:202](http://localhost/_profiler/open?file=vendor/symfony/http-kernel/Kernel.php&line=202#line202) {▶}
  [/project/my-project/public/index.php:36](http://localhost/_profiler/open?file=public/index.php&line=36#line36) {▶}
}

**Reply by martineiber** [reply]
Hi @tapli,

Try to put the value of `server_version` under single quotes, like `server_version: '8.0.32-0ubuntu0.22.04.2'`.
Otherwise you could try to change the `host` to from `localhost` to `127.0.0.1` and restart your webserver and your database server.
